_posts/develop-on-ipad.md

```diff
@@ -0,0 +1,55 @@
+---
+layout: post
+title: "An iPad as a developer box. Pros and cons"
+author: Mauro Servienti
+synopsis: "Can we use a 10-inches iPad Air as a development device? Let's find out what was my recent."
+header_image: /img/posts/develop-on-ipad/header.jpg
+tags:
+- development
+---
+
+When my 5-year-old iPad Pro fell on the floor for the third time (not accidentally, one could argue when you have 5-year-old kids. Coincidence? I don't think so!), I decided it was time for a new iPad rather than spending a lot to fix it due to an expired warranty.
+
+I bought a new iPad Air with the M1 CPU and a Magic Keyboard. No pencil, I still have the first-generation Apple Pencil, and I need to find out where it is, given that I never really used it.
+
+The main idea that justifies spending a ton of money on the keyboard is:
+
+> Given how superb [Visual Studio Code Dev Containers](https://milestone.topics.it/2021/06/11/visual-studio-code-remote-containers.html) support is, the option to use it from a browser through [GitHub Codespaces](https://github.com/features/codespaces) or [GitHub Dev](https://github.com/github/dev), and the recently introduced support for [Remote Tunnels](https://code.visualstudio.com/blogs/2022/12/07/remote-even-better)
+
+I would like to know if I can use an iPad as an on-the-go development/work device.
+
+## Specs
+
+There are a few options when buying an iPad. I went for the Air M1 one, with 256 GB of storage and a SIM card.
+
+## The good and the not so good
+
+### Connectivity
+
+Let's start with the latter. SIM card support is beneficial. I have a data plan, and not being required to tether using the phone is a plus. It saves the phone's battery life.
+
+### Portability
+
+Portability is excellent, and battery life is superb.
+
+My main concern is that the device with the keyboard is too heavy. According to my kitchen scale, it's 464 + 602 grams. A little more than a kilogram is not a big deal, but it feels too much for what's still an iPad.
+
+### Screen
+
+For prolonged periods of usage, the screen is too small. 10" is great for specific tasks, and coding is not among them.
+
+### Keyboard and trackpad
+
+The lack of the escape key drives me nuts. Muscle memory kicks in too quickly—every single time!
+
+The trackpad is small but freaking good.
+
+The great thing is that keyboard shortcuts and trackpad behaviors are what I'm used to as a macOS user.
+
+## Conclusion
+
+If it's your only option, then a 10 inches iPad can be used as a development machine. Otherwise, a bigger screen is beneficial, if not mandatory. Everything else works fine, making me think that a MacBook Air is a better compromise.
```

Comment:
```suggestion
If it's your only option, then a 10-inch iPad can be used as a development machine. Otherwise, a bigger screen is beneficial, if not mandatory. Everything else works fine, making me think that a MacBook Air is a better compromise.
```